### PR TITLE
patterns: Add a basic edge pattern that accepts a color parameter

### PR DESCRIPTION
**Pull request by tmc**
This adds a SimpleSolidPattern that demonstrates a quite simple way to accept parameters in a pattern.

### CODE REVIEW

**Comment by raldi** [comment]
Looks good, though you'll get some merge conflicts that you can solve by modifying the new LXStudio-IDE/src//main/java/titanicsend/TEApp.java to replicate the changes you made to LXStudioApp.java

**Comment by tmc** [comment]
mentioned in slack but we have indentation mismatch that I want to get sorted, will rebase and have this ready then

**Review by raldi — APPROVED** [review]

File: LXStudio-IDE/src/main/java/titanicsend/pattern/tmc/SimpleSolidEdgePattern.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+package titanicsend.pattern.tmc;
+
+import heronarts.lx.LX;
+import heronarts.lx.LXCategory;
+import heronarts.lx.color.ColorParameter;
+import heronarts.lx.color.LXColor;
+import titanicsend.pattern.TEPattern;
+
+/**
+ * SimpleSolidPattern shows a trivial pattern that accepts input to control the output color pattern.
```
Comment (raldi) [inline LXStudio-IDE/src/main/java/titanicsend/pattern/tmc/SimpleSolidEdgePattern.java:15]: It's got Edge in the name, but it colorizes all pixels, whether they're in an Edge or Panel, right? If that's on purpose, maybe rename? Or we could have a pattern that just colors the edges and use this name.

Comment (tmc) [reply LXStudio-IDE/src/main/java/titanicsend/pattern/tmc/SimpleSolidEdgePattern.java:15]: right, fixing.

Comment (tmc) [reply LXStudio-IDE/src/main/java/titanicsend/pattern/tmc/SimpleSolidEdgePattern.java:15]: changed to only set edgePoint colors

**Comment by tmc** [comment]
rebasing

**Pull request merged**